ydb/core/kqp/executer_actor/kqp_partition_helper.cpp

```diff
@@ -668,9 +769,21 @@ THashMap<ui64, TShardInfo> PrunePartitions(const NKqpProto::TKqpReadRangesSource
         TTableRange tableRange = std::holds_alternative<TSerializedCellVec>(range)
             ? TTableRange(std::get<TSerializedCellVec>(range).GetCells(), true, std::get<TSerializedCellVec>(range).GetCells(), true, true)
             : TTableRange(std::get<TSerializedTableRange>(range).ToTableRange());
+        TVector<TPartitionWithRange> readPartitions;
 
-        auto readPartitions = GetKeyRangePartitions(tableRange, stageInfo.Meta.ShardKey->GetPartitions(),
+        if (prunerConfig.BatchOperationRange) {
```

Comment:
"if (prunerConfig.BatchOperationRange) {" met this code 3 times.

Maybe make it smaller?

```
auto tableRange = GetTableRange...(...)
readPartitions = GetKeyRangePartitions(tableRange, stageInfo.Meta.ShardKey->GetPartitions(),
            keyColumnTypes);
```